src/Libraries/DSOffice/OpenXmlHelper.cs

```diff
@@ -0,0 +1,647 @@
+            {
+                throw new ArgumentException(string.Format(Properties.Resources.WorkbookNotFound, filePath));
+            }
+
+            using (var document = SpreadsheetDocument.Open(filePath, false))
+            {
+                var sheet = GetWorksheetPart(document, sheetName);
+                if (sheet == null)
+                {
+                    throw new ArgumentException(string.Format(Properties.Resources.WorksheetNotFound, sheetName));
+                }
+                var stylesheet = document.WorkbookPart.WorkbookStylesPart.Stylesheet;
+                var sharedStringTable = document.WorkbookPart.SharedStringTablePart?.SharedStringTable;
+                var rowsData = new List<object[]>();
+
+                // Takes into account holes at the beginning or in between rows and cells
+                var currentRowIndex = startRow + 1;
+                var columnCount = GetLargestColumnIndex(sheet.Worksheet.GetFirstChild<SheetData>()) - startColumn;
+                foreach (var row in sheet.Worksheet.GetFirstChild<SheetData>().Elements<Row>())
+                {
+                    if (row.RowIndex.Value < currentRowIndex)
+                    {
+                        // Skip rows until we get to where we should start
+                        continue;
+                    }
+
+                    while (row.RowIndex.Value > currentRowIndex)
+                    {
+                        // Add empty rows to fill gap. The column count used is the largest one found.
+                        rowsData.Add(new object[columnCount]);
+                        currentRowIndex++;
+                    }
+
+                    var rowData = new List<object>();
+                    var currentColumnIndex = startColumn + 1;
+                    foreach (Cell cell in row.Elements<Cell>())
+                    {
+                        var columnIndex = GetColumnIndex(cell.CellReference.Value);
+
+                        if (columnIndex < currentColumnIndex)
+                        {
+                            // Skip columns until we get to where we should start
+                            continue;
+                        }
+
+                        while (columnIndex > currentColumnIndex)
+                        {
+                            // Add empty cells to fill gap
+                            rowData.Add(null);
+                            currentColumnIndex++;
+                        }
+
+                        object value = GetCellValue(cell, sharedStringTable, stylesheet, readAsString);
+                        rowData.Add(value);
+                        currentColumnIndex++;
+                    }
+
+                    // Fill remaining columns with empty cells
+                    while (currentColumnIndex <= columnCount + startColumn)
+                    {
+                        rowData.Add(null);
+                        currentColumnIndex++;
+                    }
```

Comment:
Curious what would happen if we do not do this step?

Reply:
There are some tests that exercise this. If you have a sparse spreadsheet with lots of empty cells Dynamo is expected to still return empty values for those cells. The following would be an example, where `null` is the value to return but the cell is actually empty:
```
   1    null    null
null       2    null 
null   null       3
```
Notice that `row.Elements<Cell>` would not return empty cells, so in order to return 3 elements for the first two rows, that final loop adds the remaining values as null.

Reply:
Thanks @mmisol for the detailed explanation!